include/aspect/material_model/utilities.h

```diff
@@ -259,6 +266,88 @@ namespace aspect
                           const bool interpol,
                           const MPI_Comm &comm);
         };
+
+        /**
+        * This class reads in entropy-pressure material table and looks up for material
+        * proporties when entropy and pressure are given
+        */
+        class EntropyReader
+        {
+          public:
+
+            /**
+             * Read the table into material_lookup
```

Comment:
Mentioning material_lookup in the documentation means that if we ever change the name of the variable, we will have to remember (or more likely forget) to change this documentation as well. It is often safer to not mention the specifics of the implementation in the function documentation, and instead only describe the purpose of the function.

```suggestion
             * Read the table.
```